게시물 생성, 수정, 삭제 API

sim16065 opened the pull request:
관련 이슈
-
#이슈번호 (#20 )

📝 제목
-
[Feature] 게시물 API 개발

📋 작업 내용
-
- [ ]  게시물 생성
- [ ]  게시물 수정
- [ ]  게시물 삭제

🔧 기술 변경
-
모듈 추가
@aws-sdk/client-s3
@aws-sdk/s3-request-presigner

@aws-sdk/client-s3: AWS SDK의 S3 클라이언트 모듈로, S3 버킷과 상호작용할 때 사용됨.
@aws-sdk/s3-request-presigner: AWS S3에서 요청을 미리 서명(presign)하는 기능을 제공하는 모듈. URL 사용 시 필요.

sim16065 committed:
게시물 생성, 수정, 삭제 API

Ryu-02 commented on middlewares/validatePost.js:
```diff
@@ -0,0 +1,28 @@
+        }
+        
+        const user_id = req.session.user.id; // 세션에서 user_id 가져오기
+        req.body.user_id = user_id; // 요청 본문에 user_id 추가
```
카멜케이스를 사용하는게 어떨까용 ~~??

sim16065 replied:
오호 수정하겠습니다.

gwgw123 commented on controllers/post.ctrl.js:
```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const Post = require("../models/Post");
+const PostStorage = require('../storages/postStorage');
```
보통 개발시 파일 명에서 사용되는 띄어쓰기는 camelCase 보다 kebab-case 로 표현하더라구요 ~
저희 프로젝트에서는 보통 
![image](https://github.com/user-attachments/assets/c85c3854-d34f-4767-aba4-3001630b9f4c)
이런식으로 파일명을 구성합니다.

파일 내의 리소스/기능은 - 하이픈으로 구별하고
파일의 layer(역할) 을 구별할때는 . 을 사용하고 있습니다

여기서 적용하면 validate-post.js 라던가 post.storage.js 같이 적용해볼 수 있겠네요

하지만 팀 내 컨벤션에 맞춰서 통일만 하신다면 지금 방식도 문제가 되진 않을 것 같습니다 ~

sim16065 replied:
오 참고하겠습니다!!

gwgw123 commented on models/Post.js:
파일명 컨벤션을 정하면 좋겠네요 ~

gwgw123 commented on controllers/post.ctrl.js:
```diff
@@ -0,0 +1,88 @@
+
+        // req 객체를 createPost에 전달
+        const postData = await post.createPost(req);
+        const postId = await PostStorage.createPost(postData); // PostStorage에 저장
```
controller, model, storage 폴더를 만들어서 layer 분리를 하셨는데요

지금 controller 단에서 model 메서드를 불러오고 난 이후에 
storage 메서드를 불러서 사용하고 있는데요 이렇게 되면 mvc 패턴을 어긋나게 될 수 있고
controller 에서 storage 메서드까지 가져오다보니 controller의 책임이 생각보다 커질 수 있습니다 .

보통 mvc 패턴에서의 요청 흐름은 controller -> model ( 서비스 로직 ) -> storage( DB 접근 ) 의 흐름으로 진행됩니다.

위 코드를 예로 들면 `const postData = await post.createPost(req);` 이 실행된 후 
model 내에서 `const postId = await PostStorage.createPost(postData);` 를 실행시킬 수 있겠습니다 !

요점은 각 계층의 역할에 맞춰 코드 구성을 하고 적당한 책임 분리가 필요해 보인다고 말씀드리고 싶습니다~

역할분리, 유지보수, 확장성 등 계층 분리를 잘했을 때의 장점이 많으니 한번 찾아보시길 바랄게여

sim16065 replied:
아직 각 계층의 역할에 대해 잘 이해하지 못했던 부분이 있었는데, 자세히 설명해 주셔서 이해가 됐습니다.

controller에서 storage를 호출하는 방식이 아니라, MVC 패턴에 맞게 controller -> model -> storage의 흐름으로 책임을 분리하는 방식이 좋군요

gwgw123 replied:
네 맞습니다.
혹시 여기 pr 에서 수정하신다면 수정하시고 commit 한번 코멘트로 남겨주세요 ~

sim16065 replied:
[53b9653](https://github.com/modern-agile-team/9term-mini-back/pull/27/commits/53b9653b9b65d575ec543d92da1b3c35d32d59d8)
수정했습니다~~

sim16065 committed:
s3 폴더 추가, 폴더 리팩토링

sim16065 committed:
폴더 구조 정리 및 불필요한 파일 삭제

- 폴더별 역할 구분 및 정리
- 사용되지 않는 파일 삭제
- 미들웨어 및 컨트롤러 리팩토링
- Post 클래스 메서드 정리
- S3 이미지 업로드 코드 수정

sim16065 committed:
pull
